core/src/banking_stage/consume_worker.rs

```diff
@@ -39,12 +50,21 @@ impl ConsumeWorker {
             consumer,
             consumed_sender,
             leader_bank_notifier,
+            metrics: Arc::new(ConsumeWorkerMetrics::new(id)),
         }
     }
 
+    pub fn metrics_handle(&self) -> Arc<ConsumeWorkerMetrics> {
+        self.metrics.clone()
+    }
+
     pub fn run(self) -> Result<(), ConsumeWorkerError> {
         loop {
             let work = self.consume_receiver.recv()?;
+
+            // If the worker got here, then something happened.
+            // Flip the latch so metrics are reported next interval elapse.
+            self.metrics.latch.store(true, Ordering::Relaxed);
```

Comment:
tiny nit, `latch` is poetic `is_dirty` is practical 😏

Reply:
76ca5ea7ba:

- renamed to `has_data`
- changed "flip site" to be immediately after adding metrics **per work item**

changed flip site because it didnt' correctly handle the case where the scheduler sends work fast enough that the worker never gets through the buffer.